### ISSUE TITLE
Dailybuild Rollback Failure - entity content is too long [377716148] for the configured buffer limit [104857600].

### ISSUE THREAD

**Issue by Zawaken** [IHTSDO/snowstorm#352]
Our QA server runs into this error when it tries to do a scheduled daily build import. SNOMEDCT-NO gets locked because of this.

Full stack trace:

```
Dec 14 14:19:12 hostname java[24565]: 2021-12-14 14:19:12.222 DEBUG 24565 --- [   scheduling-1] o.s.s.dailybuild.DailyBuildService       : Daily build resources path 'file:NO/SNOMEDCT-NO'.
Dec 14 14:19:12 hostname java[24565]: 2021-12-14 14:19:12.224 DEBUG 24565 --- [   scheduling-1] o.s.s.dailybuild.DailyBuildService       : Found total builds 68 from file:NO/SNOMEDCT-NO
Dec 14 14:19:12 hostname java[24565]: 2021-12-14 14:19:12.224  INFO 24565 --- [   scheduling-1] o.s.s.dailybuild.DailyBuildService       : New daily build 2021-12-14-125652.zip found for SNOMEDCT-NO
Dec 14 14:19:12 hostname java[24565]: 2021-12-14 14:19:12.461  INFO 24565 --- [   scheduling-1] o.s.s.dailybuild.DailyBuildService       : Latest version found at 1634542601397 on MAIN/SNOMEDCT-NO.
Dec 14 14:19:20 hostname java[24565]: 2021-12-14 14:19:20.451 ERROR 24565 --- [   scheduling-1] o.s.s.d.ScheduledDailyBuildImportService : Failed to import daily build for code system SNOMEDCT-NO
Dec 14 14:19:20 hostname java[24565]: org.springframework.dao.DataAccessResourceFailureException: entity content is too long [377716148] for the configured buffer limit [104857600]; nested exception is java.lang.RuntimeException: entity content is too long [37771
Dec 14 14:19:20 hostname java[24565]: at org.springframework.data.elasticsearch.core.ElasticsearchExceptionTranslator.translateExceptionIfPossible(ElasticsearchExceptionTranslator.java:76)
Dec 14 14:19:20 hostname java[24565]: at org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate.translateException(ElasticsearchRestTemplate.java:398)
Dec 14 14:19:20 hostname java[24565]: at org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate.execute(ElasticsearchRestTemplate.java:381)
Dec 14 14:19:20 hostname java[24565]: at org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate.search(ElasticsearchRestTemplate.java:289)
Dec 14 14:19:20 hostname java[24565]: at org.springframework.data.elasticsearch.core.AbstractElasticsearchTemplate.search(AbstractElasticsearchTemplate.java:331)
Dec 14 14:19:20 hostname java[24565]: at org.snomed.snowstorm.core.data.services.SBranchService.findAllVersionsAfterOrEqualToTimestamp(SBranchService.java:99)
Dec 14 14:19:20 hostname java[24565]: at org.snomed.snowstorm.dailybuild.DailyBuildService.rollbackDailyBuildContent(DailyBuildService.java:128)
Dec 14 14:19:20 hostname java[24565]: at org.snomed.snowstorm.dailybuild.DailyBuildService.dailyBuildDeltaImport(DailyBuildService.java:96)
Dec 14 14:19:20 hostname java[24565]: at org.snomed.snowstorm.dailybuild.DailyBuildService.performScheduledImport(DailyBuildService.java:85)
Dec 14 14:19:20 hostname java[24565]: at jdk.internal.reflect.GeneratedMethodAccessor150.invoke(Unknown Source)
Dec 14 14:19:20 hostname java[24565]: at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Dec 14 14:19:20 hostname java[24565]: at java.base/java.lang.reflect.Method.invoke(Method.java:566)
Dec 14 14:19:20 hostname java[24565]: at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344)
Dec 14 14:19:20 hostname java[24565]: at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198)
Dec 14 14:19:20 hostname java[24565]: at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:774)
Dec 14 14:19:20 hostname java[24565]: at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
Dec 14 14:19:20 hostname java[24565]: at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
Dec 14 14:19:20 hostname java[24565]: at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:691)
Dec 14 14:19:20 hostname java[24565]: at org.snomed.snowstorm.dailybuild.DailyBuildService$$EnhancerBySpringCGLIB$$f1815633.performScheduledImport(<generated>)
Dec 14 14:19:20 hostname java[24565]: at org.snomed.snowstorm.dailybuild.ScheduledDailyBuildImportService.scheduledDailyBuildDeltaImport(ScheduledDailyBuildImportService.java:63)
Dec 14 14:19:20 hostname java[24565]: at jdk.internal.reflect.GeneratedMethodAccessor149.invoke(Unknown Source)
Dec 14 14:19:20 hostname java[24565]: at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Dec 14 14:19:20 hostname java[24565]: at java.base/java.lang.reflect.Method.invoke(Method.java:566)
Dec 14 14:19:20 hostname java[24565]: at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:84)
Dec 14 14:19:20 hostname java[24565]: at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)
Dec 14 14:19:20 hostname java[24565]: at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
Dec 14 14:19:20 hostname java[24565]: at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
Dec 14 14:19:20 hostname java[24565]: at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
Dec 14 14:19:20 hostname java[24565]: at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
Dec 14 14:19:20 hostname java[24565]: at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
Dec 14 14:19:20 hostname java[24565]: at java.base/java.lang.Thread.run(Thread.java:829)
Dec 14 14:19:20 hostname java[24565]: Caused by: java.lang.RuntimeException: entity content is too long [377716148] for the configured buffer limit [104857600]
Dec 14 14:19:20 hostname java[24565]: at org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate.translateException(ElasticsearchRestTemplate.java:396)
Dec 14 14:19:20 hostname java[24565]: ... 29 common frames omitted
Dec 14 14:19:20 hostname java[24565]: Caused by: java.io.IOException: entity content is too long [377716148] for the configured buffer limit [104857600]
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:828)
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.RestClient.performRequest(RestClient.java:248)
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.RestClient.performRequest(RestClient.java:251)
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.RestClient.performRequest(RestClient.java:251)
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.RestClient.performRequest(RestClient.java:235)
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1609)
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1579)
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1549)
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.RestHighLevelClient.search(RestHighLevelClient.java:1065)
Dec 14 14:19:20 hostname java[24565]: at org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate.lambda$search$11(ElasticsearchRestTemplate.java:289)
Dec 14 14:19:20 hostname java[24565]: at org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate.execute(ElasticsearchRestTemplate.java:379)
Dec 14 14:19:20 hostname java[24565]: ... 28 common frames omitted
Dec 14 14:19:20 hostname java[24565]: Suppressed: java.io.IOException: entity content is too long [377716148] for the configured buffer limit [104857600]
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:828)
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.RestClient.performRequest(RestClient.java:248)
Dec 14 14:19:20 hostname java[24565]: ... 36 common frames omitted
Dec 14 14:19:20 hostname java[24565]: Suppressed: java.io.IOException: entity content is too long [377716148] for the configured buffer limit [104857600]
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:828)
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.RestClient.performRequest(RestClient.java:248)
Dec 14 14:19:20 hostname java[24565]: ... 35 common frames omitted
Dec 14 14:19:20 hostname java[24565]: Caused by: org.apache.http.ContentTooLongException: entity content is too long [377716148] for the configured buffer limit [104857600]
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.HeapBufferedAsyncResponseConsumer.onEntityEnclosed(HeapBufferedAsyncResponseConsumer.java:76)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.nio.protocol.AbstractAsyncResponseConsumer.responseReceived(AbstractAsyncResponseConsumer.java:137)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.client.MainClientExec.responseReceived(MainClientExec.java:315)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseReceived(DefaultClientExchangeHandlerImpl.java:151)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.responseReceived(HttpAsyncRequestExecutor.java:315)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:255)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:81)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:39)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:114)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)
Dec 14 14:19:20 hostname java[24565]: ... 1 common frames omitted
Dec 14 14:19:20 hostname java[24565]: Caused by: org.apache.http.ContentTooLongException: entity content is too long [377716148] for the configured buffer limit [104857600]
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.HeapBufferedAsyncResponseConsumer.onEntityEnclosed(HeapBufferedAsyncResponseConsumer.java:76)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.nio.protocol.AbstractAsyncResponseConsumer.responseReceived(AbstractAsyncResponseConsumer.java:137)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.client.MainClientExec.responseReceived(MainClientExec.java:315)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseReceived(DefaultClientExchangeHandlerImpl.java:151)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.responseReceived(HttpAsyncRequestExecutor.java:315)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:255)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:81)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:39)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:114)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)
Dec 14 14:19:20 hostname java[24565]: ... 1 common frames omitted
Dec 14 14:19:20 hostname java[24565]: Caused by: org.apache.http.ContentTooLongException: entity content is too long [377716148] for the configured buffer limit [104857600]
Dec 14 14:19:20 hostname java[24565]: at org.elasticsearch.client.HeapBufferedAsyncResponseConsumer.onEntityEnclosed(HeapBufferedAsyncResponseConsumer.java:76)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.nio.protocol.AbstractAsyncResponseConsumer.responseReceived(AbstractAsyncResponseConsumer.java:137)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.client.MainClientExec.responseReceived(MainClientExec.java:315)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseReceived(DefaultClientExchangeHandlerImpl.java:151)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.responseReceived(HttpAsyncRequestExecutor.java:315)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:255)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:81)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:39)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:114)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
Dec 14 14:19:20 hostname java[24565]: at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)
Dec 14 14:19:20 hostname java[24565]: ... 1 common frames omitted
Dec 14 14:19:20 hostname java[24565]: 2021-12-14 14:19:20.452 DEBUG 24565 --- [   scheduling-1] o.s.s.c.d.s.i.IdentifierService          : SCTID Registration process disabled.
```

**Comment by kaicode** [comment]
This rollback operation that happens as part of the dailybuild import is attempting to select the branch/commit objects from elasticsearch since the last code system version was created. 
The response to this query is reaching over 100MB which is causing the deserialisation of that response to fail.

A workaround would be to rollback the commits since the last versioning one at a time through swagger - this way when the dailybuild process runs the selection find less results and not hit the 100MB limit.

@Zawaken Thanks for raising this following our discussion yesterday. I will fix draft a patch fix for this for you to test.

**Comment by kaicode** [comment]
@Zawaken I've committed the fix to branch [dailybuild-rollback-fix](https://github.com/IHTSDO/snowstorm/tree/dailybuild-rollback-fix). Are you able to checkout and build this branch to test the fix? 

You could build without tests (`mvn clean install -DskipTests=true`), I have run the dailybuild unit tests. 

The branch is based on the latest release (7.5.4). This rollback fix is the only change apart from some documentation updates. [Diff here](https://github.com/IHTSDO/snowstorm/compare/7.5.4...dailybuild-rollback-fix).

**Comment by kaicode** [comment]
This is fixed in the latest release (7.6.0).
